cpp/tests/io/csv_test.cpp

```diff
@@ -2194,4 +2194,90 @@ TEST_F(CsvReaderTest, DtypesMapInvalid)
   EXPECT_THROW(cudf_io::read_csv(in_opts), cudf::logic_error);
 }
 
+TEST_F(CsvReaderTest, CsvDefaultOptionsWriteReadMatch)
+{
+  auto const filepath = temp_env->get_temp_dir() + "issue.csv";
+
+  // make up some kind of dataframe
+  auto constexpr num_rows = 10;
+
+  auto int8_column = []() {
+    auto values = random_values<int8_t>(num_rows);
+    return column_wrapper<int8_t>(values.begin(), values.end());
+  }();
+  auto int16_column = []() {
+    auto values = random_values<int16_t>(num_rows);
+    return column_wrapper<int16_t>(values.begin(), values.end());
+  }();
```

Comment:
Now that we have a bunch more columns, we can see there's a lot of repetition in how they're defined. In this case it looks like we can consolidate a lot of this using template metaprogramming.

```c++
template<typename T>
create_random_column_wrapper(int32_t num_rows)
{
  auto values = random_values<T>(num_rows);
  return column_wrapper<T>(values.begin(), values.end());
}
```

```suggestion
  auto int8_column = create_random_column_wrapper<int8_t>(num_rows);
  auto int16_column = create_random_column_wrapper<int16_t>(num_rows);
```